### PR TITLE
Do not allow password to be same as email/username

### PR DESCRIPTION
**Pull request by knolleary**
## Description

This adds an additional requirement for passwords to not match a user's email/username.

I've tried to standardise the various forms which allows a user to enter/change their password to be consistent in feedback.

Fixes https://github.com/FlowFuse/security/issues/47

### CODE REVIEW

**Comment by knolleary** [comment]
Looking at test failures...

**Comment by knolleary** [comment]
I enabled password strength checks on the reset-password route - but forgot to update the tests to use a strong enough password to pass

File: forge/db/controllers/User.js [hunk 1]
```diff
@@ -39,9 +44,14 @@ module.exports = {
     },
 
     resetPassword: async function (app, user, newPassword) {
-        // if (zxcvbn(newPassword.score < 3)) {
-        //     throw new Error('Password Too Weak')
-        // }
+        if (zxcvbn(newPassword).score < 3) {
```
Comment (hardillb) [inline forge/db/controllers/User.js:47]:
```suggestion
        if (zxcvbn(newPassword).score < 2) {
```

Everywhere else in the code the strength is tested to `< 2`

**Review by hardillb — CHANGES_REQUESTED** [review]
Looks ok, but the reset rule is stronger than the rest ( `score < 3` vs  `score < 2` everywhere else.

File: forge/db/controllers/User.js (hunk 1 above)
Comment (knolleary) [reply forge/db/controllers/User.js:47]: I just uncommented your code ;)

**Pull request merged**